src/Action/SyliusConvertAction.php

```diff
@@ -42,6 +55,11 @@ public function execute($request)
 
         $model = ArrayObject::ensureArrayObject($payment->getDetails());
 
+        //If there is a new token, clear all details
+        if (isset($model['success_url']) && basename($model['success_url']) !== $request->getToken()->getHash()) {
+            $model = new ArrayObject();
+        }
+
```

Comment:
Is this allow multiple payments on the same order ? If it does, it's a good idea.

Reply:
I didn't test with multiple payments yet, my project doesn't support it

Reply:
Ok I will retest it while finishing my tests on another gateway : https://github.com/Prometee/PayumStripeCheckoutSession